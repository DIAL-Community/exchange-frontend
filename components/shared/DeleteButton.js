import { BsTrash } from 'react-icons/bs'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'

const DeleteButton = ({ type = 'button', onClick, href, className }) => {
  const innerHtml =
    <>
      <BsTrash className='text-sm'/>
      <span className='text-sm'>
        <FormattedMessage id='app.delete' />
      </span>
    </>

  const style = 'delete-button gap-x-1.5'

  return (
    type === 'button' ? (
      <button
        type='button'
        onClick={onClick}
        className={classNames(className, style)}
        data-testid='delete-button'
      >
        {innerHtml}
      </button>
    ) : type === 'link' && (
      <a
        href={href}
        className={classNames(className, style)}
        data-testid='delete-link'
      >
        {innerHtml}
      </a>
    )
  )
}

export default DeleteButton
