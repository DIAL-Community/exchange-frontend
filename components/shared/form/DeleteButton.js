import { FaRegTrashCan } from 'react-icons/fa6'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'

const DeleteButton = ({ type = 'button', title = 'app.delete', onClick, href, className }) => {
  const innerHtml =
    <>
      <FaRegTrashCan className='text-sm'/>
      <span className='text-sm'>
        <FormattedMessage id={title} />
      </span>
    </>

  const style = 'delete-button gap-x-1.5'

  return (
    type === 'button'
      ? (
        <button type='button' onClick={onClick} className={classNames(className, style)}>
          {innerHtml}
        </button>
      )
      : type === 'link' && (
        <a href={href} className={classNames(className, style)}>
          {innerHtml}
        </a>
      )
  )
}

export default DeleteButton
