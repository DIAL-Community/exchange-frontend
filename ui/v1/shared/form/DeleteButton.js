import { FaTrashAlt } from 'react-icons/fa6'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'

const DeleteButton = ({ type = 'button', title = 'app.delete', onClick, href, className }) => {
  const innerHtml =
    <>
      <FaTrashAlt className='text-sm'/>
      <span className='text-sm'>
        <FormattedMessage id={title} />
      </span>
    </>

  const style = 'delete-button gap-x-1.5'

  return (
    type === 'button' ? (
      <button
        type='button'
        onClick={onClick}
        className={classNames(className, style)}
        data-testid='delete-button'
      >
        {innerHtml}
      </button>
    ) : type === 'link' && (
      <a
        href={href}
        className={classNames(className, style)}
        data-testid='delete-link'
      >
        {innerHtml}
      </a>
    )
  )
}

export default DeleteButton
