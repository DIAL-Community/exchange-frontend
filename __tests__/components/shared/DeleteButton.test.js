import { fireEvent } from '@testing-library/react'
import DeleteButton from '../../../components/shared/DeleteButton'
import { mockRouterImplementation, render } from '../../test-utils'

// Mock next-router calls.
jest.mock('next/dist/client/router')

describe('Unit test for the DeleteButton component.', () => {
  const DELETE_BUTTON_TEST_ID = 'delete-button'
  const DELETE_LINK_TEST_ID = 'delete-link'

  beforeAll(mockRouterImplementation)

  test('Should match snapshot - type === "button".', () => {
    const { container, queryByTestId } = render(<DeleteButton />)
    expect(container).toMatchSnapshot()
    expect(queryByTestId(DELETE_BUTTON_TEST_ID)).toHaveProperty('type', 'button')
    expect(queryByTestId(DELETE_BUTTON_TEST_ID)).toHaveTextContent('Delete')
    expect(queryByTestId(DELETE_LINK_TEST_ID)).toBeNull()
  })

  test('Should match snapshot - type === "link".', () => {
    const { container, queryByTestId } = render(<DeleteButton type='link' href='http://test/' />)
    expect(container).toMatchSnapshot()
    expect(queryByTestId(DELETE_LINK_TEST_ID)).toHaveProperty('href', 'http://test/')
    expect(queryByTestId(DELETE_LINK_TEST_ID)).toHaveTextContent('Delete')
    expect(queryByTestId(DELETE_BUTTON_TEST_ID)).toBeNull()
  })

  test('Should call "onClick" on "Delete" button click - type === "button".', () => {
    const mockOnClick = jest.fn()
    const { getByTestId } = render(<DeleteButton onClick={mockOnClick}/>)
    fireEvent.click(getByTestId(DELETE_BUTTON_TEST_ID))
    expect(mockOnClick).toBeCalled()
  })
})
