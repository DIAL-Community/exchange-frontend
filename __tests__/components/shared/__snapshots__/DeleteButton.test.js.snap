// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the DeleteButton component. Should match snapshot - type === "button". 1`] = `
<div>
  <div
    class="Toastify"
  />
  <button
    class="delete-button gap-x-1.5"
    data-testid="delete-button"
    type="button"
  >
    <svg
      class="text-sm"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
      />
    </svg>
    <span
      class="text-sm"
    >
      Delete
    </span>
  </button>
</div>
`;

exports[`Unit test for the DeleteButton component. Should match snapshot - type === "link". 1`] = `
<div>
  <div
    class="Toastify"
  />
  <a
    class="delete-button gap-x-1.5"
    data-testid="delete-link"
    href="http://test/"
  >
    <svg
      class="text-sm"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M268 416h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12zM432 80h-82.41l-34-56.7A48 48 0 0 0 274.41 0H173.59a48 48 0 0 0-41.16 23.3L98.41 80H16A16 16 0 0 0 0 96v16a16 16 0 0 0 16 16h16v336a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128h16a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM171.84 50.91A6 6 0 0 1 177 48h94a6 6 0 0 1 5.15 2.91L293.61 80H154.39zM368 464H80V128h288zm-212-48h24a12 12 0 0 0 12-12V188a12 12 0 0 0-12-12h-24a12 12 0 0 0-12 12v216a12 12 0 0 0 12 12z"
      />
    </svg>
    <span
      class="text-sm"
    >
      Delete
    </span>
  </a>
</div>
`;
